extensions/smallrye-reactive-messaging-kafka/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/kafka/HibernateOrmStateStore.java

```diff
@@ -72,19 +72,29 @@ public Uni<Map<TopicPartition, ProcessingState<?>>> fetchProcessingState(Collect
             Object[] ids = partitions.stream()
                     .map(tp -> new CheckpointEntityId(consumerGroupId, tp))
                     .toArray(Object[]::new);
-            return Vertx.currentContext().executeBlocking(Uni.createFrom().item(() -> {
+            return Vertx.currentContext().executeBlocking(Uni.createFrom().emitter(emitter -> {
                 List<CheckpointEntity> fetched = new ArrayList<>();
+                Transaction tx = null;
                 try (Session session = sf.openSession()) {
+                    tx = session.beginTransaction();
                     for (Object id : ids) {
                         CheckpointEntity entity = session.find(stateType, id);
                         if (entity != null) {
                             fetched.add(entity);
                         }
                     }
+                    session.flush();
+                    tx.commit();
+                    emitter.complete(fetched.stream().filter(e -> e != null && CheckpointEntity.topicPartition(e) != null)
+                            .collect(Collectors.toMap(CheckpointEntity::topicPartition,
+                                    e -> new ProcessingState<>(e, e.offset))));
```

Comment:
An exception cannot be thrown there? If so, you might end up with the transaction not being null and already committed and trying to rollback in the catch. Maybe you should nullify `tx` after committing?

Reply:
That's true. I'll move that code up before the commit.